runtime/lua/vim/_editor.lua

```diff
@@ -329,7 +329,13 @@ do
       vim.api.nvim_command(('echo "%s"'):format(dots))
     end
     if is_last_chunk then
+      if startpos then
+        vim.fn.setpos("'[", startpos)
+        startpos = nil
```

Comment:
Why check `startpos` instead of `phase == 2 or phase == 3` ? 

We can call `setpos()` always during phase 2/3. That ensures the `'[` mark is always correct during a stream, instead of being temporarily wrong until the last chunk.

```suggestion
      if phase == 2 or phase == 3 then
        vim.fn.setpos("'[", assert(startpos))
        startpos = nil
```